### PR TITLE
fix: fix dialyzer error for systemd:set_status/1

### PR DESCRIPTION
**Pull request by joaohf**
Hello!

This fixes the issue #33. It is about dialyzer checks. 

Thanks.

### CODE REVIEW

File: src/systemd.erl [hunk 1]
```diff
@@ -195,8 +195,8 @@ ready(Status) ->
 
 -spec set_status(Statuses) -> supervisor:child_spec() when
     Statuses :: MapStatuses | ListStatuses,
-    ListStatuses :: [{up | down, state()}],
-    MapStatuses :: #{up := state(), down := state()}.
+    ListStatuses :: [{up | down, [state()]}],
+    MapStatuses :: #{up => [state()], down => [state()]}.
```
Comment (hauleth) [inline src/systemd.erl:199]: I think it should be:

```suggestion
    ListStatuses :: [{up | down, State}],
    MapStatuses :: State, down => State},
    State :: state() | [state()].
```

Comment (joaohf) [reply src/systemd.erl]: Agreed.

**Comment by joaohf** [comment]
Hi @hauleth 

Let me know if any further changes are needed.

Thanks.

**Comment by hauleth** [comment]
💚 💙 💜 💛 ❤️

**Pull request merged**